src/node.rs

```diff
@@ -773,6 +866,57 @@ impl<'a, V: 'a> Iterator for Iter<'a, V> {
     }
 }
 
+/// A mutable iterator which traverses the nodes in a tree, in depth first order.
+///
+/// The first element of an item is the level of the traversing node.
+#[derive(Debug)]
+pub struct IterMut<'a, V: 'a> {
+    stack: Vec<(usize, &'a mut Node<V>)>,
+}
+
+/// A reference to an immediate node (without child or sibling) with its
+/// label and a mutable reference to its value, if present.
+pub struct NodeMutRef<'a, V: 'a> {
+    label: &'a [u8],
+    value: Option<&'a mut V>,
+}
+impl<'a, V: 'a> NodeMutRef<'a, V> {
+    /// Makes a new reference to a node's label and mutable value.
+    pub fn new(label: &'a [u8], value: Option<&'a mut V>) -> Self {
+        NodeMutRef { label, value }
+    }
+
+    /// Returns the label of the referenced node.
+    pub fn label(&self) -> &'a [u8] {
+        self.label
+    }
+
+    /// Returns the optional reference to the value stored within the referenced node.
+    pub fn take_value_mut(self) -> Option<&'a mut V> {
```

Comment:
IMO, the current name seems a bit confusing because it doesn't actually take ownership of the value (i.e., the value will not be removed from the node after calling this method).
So how about renaming it as follows?
```suggestion
    pub fn into_value_mut(self) -> Option<&'a mut V> {
```

Reply:
Gotcha, that makes sense why it's better to just name it `into_*`. I've renamed the function.